### PR TITLE
161780579 Users can like or dislike comments

### PR DESCRIPTION
**Pull request by innocentEdosa**
#### Description
This PR adds the functionality for liking and disliking articles

#### Type of change
Feature
 - create comment likes model
- set up likeCommentController
- write tests for comment like-dislike functionality

#### How Has This Been Tested?
Clone the repo
checkout out to feature/161780579-user-like-dislike-comments
run NPM start
test with postman

#### Pivotal tracker Id
#161780579

Blank Diagram (1)

### CODE REVIEW

File: src/middlewares/validator/validator.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ const body = joi.string().required().error(() => 'Body is required');
 const tagList = joi.array().required().error(() => 'tagList must be an array ');
 const plainText = joi.string().required().error(() => 'Plain text is required');
 
-const articleId = joi.string().guid().required().error(() => 'We do not know the article you are trying to rate. Please provide a valid article id');
+const articleId = joi.string().guid().required().error(() => 'We do not know the article you are referring to rate. Please provide a valid article id');
+const commentId = joi.string().guid().required().error(() => 'We do not know the comment you are referring to. Please provide a comment id');
```
Comment (kayroy247) [inline src/middlewares/validator/validator.js]: You have done a nice job with the way you implemented validation here but the 2 lines of code above are long. You may need to make them shorter to improve the readability.

**Review by kayroy247 — CHANGES_REQUESTED** [review]

File: src/middlewares/validator/validator.js (hunk 1 above)
Comment (innocentEdosa) [reply src/middlewares/validator/validator.js]: Thanks for the review Kayroy247

Readability is an important part of coding.
I would implement the suggested changes now

Thanks again

**Review by kayroy247 — APPROVED** [review]

**Comment by Bobjayafam** [comment]
LGTM

File: src/routes/api/comments/likeCommentController.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import models from '../../../models';
+
+const { comment } = models;
```
[the remark this answers is not in the document]
Comment (fuchodeveloper) [reply src/routes/api/comments/likeCommentController.js]: Please resolve HoundCI violation.

File: src/routes/api/comments/likeCommentController.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      type: models.sequelize.QueryTypes.SELECT
+    });
+
+    likeResult.likes = likeResult.likes ? Number(likeResult.likes) : 0;
+    likeResult.dislikes = likeResult.dislikes ? Number(likeResult.dislikes) : 0;
+
+    return response.status(200).send({ status: 200, data: likeResult });
+  } catch (err) {
+    if (err.name === 'SequelizeForeignKeyConstraintError') {
+      response.status(404).json({ status: 404, error: 'Comment does not exist' })
```
[the remark this answers is not in the document]
Comment (fuchodeveloper) [reply src/routes/api/comments/likeCommentController.js]: Please resolve HoundCI violation.

**Pull request merged**